xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/DefaultNotificationFilterPreferenceManager.java

```diff
@@ -169,16 +169,22 @@ public void saveFilterPreferences(DocumentReference user, Set<NotificationFilter
 
     @Override
     public void deleteFilterPreference(DocumentReference user, String filterPreferenceId) throws NotificationException
+    {
+        deleteFilterPreferences(user, Set.of(filterPreferenceId));
+    }
+
+    @Override
+    public void deleteFilterPreferences(DocumentReference user, Set<String> filterPreferenceIds)
+        throws NotificationException
     {
         try {
             for (NotificationFilterPreferenceProvider provider
-                    : componentManager.<NotificationFilterPreferenceProvider>getInstanceList(
-                    NotificationFilterPreferenceProvider.class)) {
-                provider.deleteFilterPreference(user, filterPreferenceId);
+                : componentManager.<NotificationFilterPreferenceProvider>getInstanceList(
+                NotificationFilterPreferenceProvider.class)) {
+                provider.deleteFilterPreferences(user, filterPreferenceIds);
             }
-
         } catch (ComponentLookupException e) {
-            logger.info("Failed to remove the user filter preference [{}].", filterPreferenceId, e);
+            logger.info("Failed to remove the user filter preference [{}].", filterPreferenceIds, e);
```

Comment:
You might want to update a bit the message (plural, and you already get `[]` around a `Set.toString`).